config/routes.rb

```diff
@@ -0,0 +1,22 @@
+Rails.application.routes.draw do
+  devise_for :users
+  
+  devise_scope :user do
+  authenticated do
+    root to: 'groups#index'
+  end
+
+  unauthenticated do
+    root to: 'welcomepages#index', as: 'unauthenticated_user_root'
+  end
+end
+  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+
+  # Defines the root path route ("/")
+  # root "articles#index"
+
+  resources :users, only: %i[index show]
+  resources :groups, only: %i[index show new create destroy] do
+    resources :entities, only: %i[index show new create destroy]
```

Comment:
- [ ] Kindly add only the necessary routes. You didn't implement the delete action